shenyu-client/shenyu-client-motan/src/main/java/org/apache/shenyu/client/motan/MotanServiceEventListener.java

```diff
@@ -89,52 +88,52 @@ public MotanServiceEventListener(final PropertiesConfig clientConfig, final Shen
-        if (superPath.contains("*") && Objects.nonNull(beanShenyuClient)) {
-            Method[] methods = ReflectionUtils.getDeclaredMethods(clazz);
-            for (Method method : methods) {
-                publisher.publishEvent(buildMetaDataDTO(clazz, service, beanShenyuClient, method,
-                        buildRpcExt(method, timeout), superPath));
-            }
-            return;
-        }
-        Method[] methods = ReflectionUtils.getUniqueDeclaredMethods(clazz);
-        for (Method method : methods) {
-            ShenyuMotanClient shenyuMotanClient = AnnotatedElementUtils.findMergedAnnotation(method, ShenyuMotanClient.class);
-            if (Objects.nonNull(shenyuMotanClient)) {
-                publisher.publishEvent(buildMetaDataDTO(clazz, service,
-                        shenyuMotanClient, method, buildRpcExt(method, timeout), superPath));
-            }
+    @Override
+    protected Map<String, Object> getBeans(final ApplicationContext context) {
+        return context.getBeansWithAnnotation(ShenyuMotanClient.class);
+    }
+
+    @Override
+    protected URIRegisterDTO buildURIRegisterDTO(final ApplicationContext context, final Map<String, Object> beans) {
+        return URIRegisterDTO.builder()
+                .contextPath(contextPath)
+                .appName(appName)
+                .rpcType(RpcTypeEnum.MOTAN.getName())
+                .host(host)
+                .port(Integer.parseInt(port))
+                .build();
+    }
+
+    @Override
+    protected String buildApiSuperPath(final Class<?> clazz, final ShenyuMotanClient shenyuMotanClient) {
+        if (Objects.nonNull(shenyuMotanClient) && !StringUtils.isBlank(shenyuMotanClient.path())) {
+            return shenyuMotanClient.path();
         }
+        return "";
     }
 
-    private MetaDataRegisterDTO buildMetaDataDTO(final Class<?> clazz, final MotanService service,
-                                                 final ShenyuMotanClient shenyuMotanClient,
-                                                 final Method method, final String rpcExt, final String superPath) {
-        String appName = this.appName;
-        String path = superPath.contains("*") ? pathJoin(contextPath, superPath.replace("*", ""), method.getName()) : pathJoin(contextPath, superPath, shenyuMotanClient.path());
+    @Override
+    protected Class<ShenyuMotanClient> getAnnotationType() {
+        return ShenyuMotanClient.class;
+    }
+
+    @Override
+    protected MetaDataRegisterDTO buildMetaDataDTO(final Object bean, final ShenyuMotanClient shenyuMotanClient, final String superPath, final Class<?> clazz, final Method method) {
+        Integer timeout = Optional.ofNullable(((BasicServiceConfigBean) applicationContext.getBean(BASE_SERVICE_CONFIG)).getRequestTimeout()).orElse(1000);
+        MotanService service = clazz.getAnnotation(MotanService.class);
```

Comment:
use `AnnotatedElementUtils#findMergedAnnotation` to replace `Class#getAnnotation` for better scalability.

Reply:
Can you give a reason for using `AnnotatedElementUtils#getMergedAnnotation`?

Reply:
Should I use AnnotatedElementUtils#findMergedAnnotation to replace Class#getAnnotation for better scalability?

Reply:
> Can you give a reason for using `AnnotatedElementUtils#getMergedAnnotation`?

Oh,I got it, I should use findMergedAnnotation... I use wrong function..